PLEP-0007.rst

```diff
@@ -0,0 +1,364 @@
+|                   | | Nicholas A. Murphy                        |
++-------------------+---------------------------------------------+
+| contact email     | eeverson@ucla.edu                           |
++-------------------+---------------------------------------------+
+| date created      | 2019-10-27                                  |
++-------------------+---------------------------------------------+
+| date last revised | 2020-05-12                                  |
++-------------------+---------------------------------------------+
+| type              | Standard                                    |
++-------------------+---------------------------------------------+
+| status            | Accepted                                    |
++-------------------+---------------------------------------------+
+| DOI               | `10.5281/zenodo.3774573`_                   |
+|                   |                                             |
++-------------------+---------------------------------------------+
+
+**Contents**
+
+* `Abstract`_
+* `Detailed Description`_
+
+  * `Why?`_
+  * `Top-Level Sub-Packages`_
+
+* `Extensible Packages are Exempt`_
+* `Implementation`_
+* `Issues, Pull Requests, and Branches`_
+* `Backward Compatibility`_
+* `Decision Rationale`_
+
+Abstract
+========
+
+This PLEP defines the top-level structure of ``PlasmaPy`` (i.e.
+``plasmapy.<subpackage>``).  Its intent is to name the top-level
+sub-packages and define the general scope of each package.
+Sub-package details are left to be defined in package focused PLEPs,
+which should *NOT* conflict with this PLEP.  Defining the top-level
+structure should:
+
+  #. keep the top-level namespace from getting bloated
+  #. help developers decided where new code should be placed
+  #. help navigate users to the functionality they need
+
+If new code does NOT fall within the current framework after a thorough and
+exhaustive review, then this PLEP needs to be modified/updated accordingly
+before any new top-level sub-packages are created.
+
+Detailed Description
+====================
+
+Why?
+----
+
+By defining the top-level sub-package namespace developers are better
+directed to where code should be added, the ``PlasmaPy``
+`Coordinating Committee
+<https://github.com/PlasmaPy/PlasmaPy-PLEPs/blob/master/PLEP-0003.rst>`_
+can better manage package bloat, and users are better directed to the code
+they need.
+
+Top-Level Sub-Packages
+----------------------
+
+.. _`PlasmaPy`: https://www.plasmapy.org/
+.. _`NRL Plasma Formulary`: https://www.nrl.navy.mil/ppd/content/nrl-plasma-formulary
+
+The following sub-packages are outlined in this PLEP, with brief synopses given in
+the tables below.  The details of each package are left to be fully defined by their
+respective package PLEPs.
+
+  #. `addons`_
+  #. `analysis`_
+  #. `diagnostics`_
+  #. `dispersion`_
+  #. `formulary`_
+  #. `particles`_
+  #. `plasma`_
```

Comment:
I forget if we decided to have a separate fields subpackage or if we wanted to keep fields a part of plasma subpackage.

Reply:
Yes, as the [*scope* for `plasmapy.plasma`](https://github.com/rocco8773/PlasmaPy-PLEPs/blob/PLEP7_top_level_package_structure/PLEP-0007.rst#plasma) details, anything that is part of defining the plasma environment (include fields) would be placed within `plasmapy.plasma`.

Reply:
That is what confused me. I would typically think of field solvers as being complicated enough to be a sub-package by themselves, given that the field equations will be very different based on what approximation (Boltzmann, Vlasov, Fluid, GK...) we're working with. 

However, now I remember part of our discussion at DPP. Part of the discussion was whether we would need/want to reproduce a lot of EM field solver libraries or if would want to have just enough for our purposes. Most people on the table wanted to keep the scope limited and within the plasma sub-package, so I had agreed to this option. 

Now that I remember the details, I will approve the pull request.